bot.py

```
@@ -538,18 +609,26 @@ async def send_temporary_notification(self, thread, reply_info, deletion_success
             # Schedule automatic deletion after 8 seconds (if we have permission)
             if deletion_successful:
                 self.logger.debug(
                     f"Sent temporary notification to {user} in #{channel.name}, "
                     f"message will auto-delete in 8 seconds"
                 )
-                await asyncio.sleep(8)
-                await notification_message.delete()
-                self.logger.debug(f"Auto-deleted notification message {notification_message.id} in #{channel.name}")
+                # Defer the auto-delete so the main flow returns promptly and
+                # we don't hold the process_reply task open for 8 seconds.
+                asyncio.create_task(self._delete_after(notification_message, 8))
```

Ruff (0.15.13)

[warning] Store a reference to the return value of `asyncio.create_task`

(RUF006)